lib/presentation/widgets/list_characters.dart

```diff
@@ -30,36 +32,10 @@ class _ListCharactersState extends State<ListCharacters> {
 
   @override
   Widget build(BuildContext context) {
-    return BlocBuilder<HomeBloc, HomeState>(
-      builder: (context, state) {
-        Widget listViewBuilder = _getListViewBuilder(state.characters, state.hasReachedMax, state.loading, state.error);
-        if (state.loading) {
-          return state.characters.isEmpty ? const CenterLoader() : listViewBuilder;
-        }
-        if (state.error != null) {
-          Widget errorWidget = PageError(
-            onRetry: () {
-              context.read<HomeBloc>().add(ReadyForData());
-            },
-          );
-          return state.characters.isEmpty ? errorWidget : listViewBuilder;
-        }
-        return listViewBuilder;
-      },
-    );
-  }
-
-  @override
-  void dispose() {
-    _scrollController
-      ..removeListener(_onScroll)
-      ..dispose();
-    super.dispose();
-  }
-
-  //TODO Кандидат в отдельный виджет вместо того чтобы быть методом
-  //TODO в этом виджете методе не должно быть параметра loading
-  Widget _getListViewBuilder(List<CharacterViewData> characters, bool hasReachedMax, bool loading, String? error) {
+    var characters = widget.characters;
+    bool hasReachedMax = widget.hasReachedMax;
+    bool loading = widget.loading;
+    String? error = widget.error;
```

Comment:
Looks like you are not using this string from error, but we discussed that we will show different messages based on the error type